src/notation/view/noteinputbarmodel.cpp

```diff
@@ -558,10 +598,227 @@ MenuItem NoteInputBarModel::makeActionItem(const ActionItem& action, const std::
+        makeAction("chord-b")
+    };
+
+    return items;
+}
+
+MenuItemList NoteInputBarModel::intervalsItems() const
+{
+    MenuItemList items {
+        makeAction("interval1"),
+        makeAction("interval2"),
+        makeAction("interval3"),
+        makeAction("interval4"),
+        makeAction("interval5"),
+        makeAction("interval6"),
+        makeAction("interval7"),
+        makeAction("interval8"),
+        makeAction("interval9"),
+        makeSeparator(),
+        makeAction("interval-2"),
+        makeAction("interval-3"),
+        makeAction("interval-4"),
+        makeAction("interval-5"),
+        makeAction("interval-6"),
+        makeAction("interval-7"),
+        makeAction("interval-8"),
+        makeAction("interval-9")
+    };
+
+    return items;
+}
+
+MenuItemList NoteInputBarModel::measuresItems() const
+{
+    MenuItemList items {
+        makeAction("insert-measure"),
+        makeAction("insert-measures"),
+        makeSeparator(),
+        makeAction("append-measure"),
+        makeAction("append-measures")
+    };
+
+    return items;
+}
+
+MenuItemList NoteInputBarModel::framesItems() const
+{
+    MenuItemList items {
+        makeAction("insert-hbox"),
+        makeAction("insert-vbox"),
+        makeAction("insert-textframe"),
+        makeSeparator(),
+        makeAction("append-hbox"),
+        makeAction("append-vbox"),
+        makeAction("append-textframe")
+    };
+
+    return items;
+}
+
+MenuItemList NoteInputBarModel::textItems() const
+{
+    MenuItemList items {
+        makeAction("title-text"),
+        makeAction("subtitle-text"),
+        makeAction("composer-text"),
+        makeAction("poet-text"),
+        makeAction("part-text"),
+        makeSeparator(),
+        makeAction("system-text"),
+        makeAction("staff-text"),
+        makeAction("expression-text"),
+        makeAction("rehearsalmark-text"),
+        makeAction("instrument-change-text"),
+        makeAction("fingering-text"),
+        makeSeparator(),
+        makeAction("sticking-text"),
+        makeAction("chord-text"),
+        makeAction("roman-numeral-text"),
+        makeAction("nashville-number-text"),
+        makeAction("lyrics"),
+        makeAction("figured-bass"),
+        makeAction("tempo")
+    };
+
+    return items;
+}
+
+MenuItemList NoteInputBarModel::linesItems() const
+{
+    MenuItemList items {
+        makeAction("add-slur"),
+        makeAction("add-hairpin"),
+        makeAction("add-hairpin-reverse"),
+        makeAction("add-8va"),
+        makeAction("add-8vb"),
+        makeAction("add-noteline")
+    };
+
+    return items;
+}
+
+bool NoteInputBarModel::isNeedShowSubitemsByClick(const ActionCode& actionCode) const
+{
+    if (isNoteInputModeAction(actionCode)) {
+        return false;
+    }
+
+    return true;
+}
+
+void NoteInputBarModel::notifyAboutTupletItemChanged()
+{
+    int tupletItemIndex = itemIndex(TUPLET_ACTION_CODE);
+    if (tupletItemIndex == -1) {
```

Comment:
It is better to add a static constant for this number (INVALID_INDEX), you use it in many places